lualib/resty/events/worker.lua

```diff
@@ -243,21 +244,23 @@ function _M:communicate(premature)
             sleep(0)
 
             ::continue::
-        end -- while not exiting
-    end)  -- events_thread
+        end -- while not exiting or restarting
+    end) -- events_thread
 
     local ok, err, perr = wait(write_thread, read_thread, events_thread)
 
-    kill(write_thread)
     kill(read_thread)
-    kill(events_thread)
-
-    self._connected = nil
+    kill(write_thread)
```

Comment:
After we kill these two threads, the `self._connected` should be nil now.

Reply:
> After we kill these two threads, the `self._connected` should be nil now.

It is just somekind of `testing only` variable and it doesn't matter that it is down there. Looks cleaner when moved down.